src/tiled/automapper.cpp

```diff
@@ -749,7 +797,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 if (listNo.isEmpty()) {
                     if (matchListYes)
                         continue;
-                    if (!ruleDefinedListYes && !cells.contains(c1))
```

Comment:
This change would require an update to the comments, in particular this part:
```
 *      Exception, when having only the listYes:
 *      if at the examined position there are no tiles within all Layers
 *      of the listYes, all tiles except all used tiles within
 *      the layers of that list are considered good.
 *
 *      This exception was added to have a better functionality
 *      (need of less layers.)
```
And apparently, this makes some case worse. Essentially, leaving a tile empty in the `listYes` is taken to mean that there should not be any tile used elsewhere in that same rule (when there are no explicit exclusion rules). Without that logic you need to fill the `listNo` with all those tiles to get the same functionality. That seems like a nice shortcut, but it is indeed rather unexpected. Maybe @stefanbeller could weigh in on this?